packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -467,16 +467,20 @@ class HotRunner extends ResidentRunner {
       final String entryPath = fs.path.relative(mainPath, from: projectRootPath);
       if (benchmarkMode)
         vmReloadTimer.start();
-      Map<String, dynamic> reloadReport;
-      final List<Future<Map<String, dynamic>>> reloadReports = <Future<Map<String, dynamic>>>[];
+      final List<Future<Map<String, dynamic>>> reloadReportFutures = <Future<Map<String, dynamic>>>[];
       for (FlutterDevice device in flutterDevices) {
         final List<Future<Map<String, dynamic>>> reports = device.reloadSources(
           entryPath,
           pause: pause
         );
-        reloadReports.addAll(reports);
+        reloadReportFutures.addAll(reports);
       }
-      reloadReport = (await Future.wait(reloadReports)).first;
+      final List<Map<String, dynamic>> reloadReports = await Future.wait(reloadReportFutures);
+      if (reloadReports.isEmpty) {
```

Comment:
Can we move this check from `reloadReport` to `reloadReportFutures`?
If `reloadReport` is empty it means that `reloadReportFutures` was empty too.